src/boot/efi/boot.c

```diff
@@ -1651,6 +1651,51 @@ static EFI_STATUS image_start(EFI_HANDLE parent_image, const Config *config, con
                 loaded_image->LoadOptionsSize = (StrLen(loaded_image->LoadOptions)+1) * sizeof(CHAR16);
 
 #ifdef SD_BOOT_LOG_TPM
+                UINTN i, j;
+                UINTN length;
+                CHAR16* begin, *end;
+
+                EFI_FILE* root_dir = LibOpenRoot(entry->device);
+                if (!root_dir) {
+                        Print(L"Unable to open root directory: %r ", err);
+                        uefi_call_wrapper(BS->Stall, 1, 3 * 1000 * 1000);
```

Comment:
This is the same statement that is used throughout boot.c, it salls, i.e. waits for 3 seconds to allow the user to read the error message